content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md

```diff
@@ -0,0 +1,410 @@
+import tabulate # Required for pandas.to_markdown()
+
+def explain_and_analyze():
+  print('Use SQL EXPLAIN and ANALYZE to view query plan information.')
+
+  # Instantiate an InfluxDB client.
+  client = InfluxDBClient3(token = f"DATABASE_TOKEN",
+                          host = f"{{< influxdb/host >}}",
+                          database = f"DATABASE_NAME")
+
+  sql_explain = '''EXPLAIN SELECT *
+        FROM home
+        WHERE time >= now() - INTERVAL '90 days'
+        ORDER BY time'''
+
+  table = client.query(sql_explain)
+  df = table.to_pandas()
+
+  sql_explain_analyze = '''EXPLAIN ANALYZE SELECT *
+      FROM home
+      WHERE time >= now() - INTERVAL '90 days'
+      ORDER BY time'''
+
+  table = client.query(sql_explain_analyze)
+
+  # Combine the Dataframes and output the plan information.
+  df = pd.concat([df, table.to_pandas()])
+
+  assert df.shape == (3, 2) and df.columns.to_list() == ['plan_type', 'plan']
+  print(df[['plan_type', 'plan']].to_markdown(index=False))
+
+  client.close()
+
+explain_and_analyze()
+```
+{{% /code-placeholders %}}
+
+Replace the following:
+
+- {{% code-placeholder-key %}}`DATABASE_NAME`{{% /code-placeholder-key %}}: your {{% product-name %}} database
+- {{% code-placeholder-key %}}`DATABASE_TOKEN`{{% /code-placeholder-key %}}: a [database token](/influxdb/cloud-dedicated/admin/tokens/) with sufficient permissions to the specified database
+
+The output is similar to the following:
+
+```markdown
+| plan_type         | plan                                                                                                                                         |
+|:------------------|:---------------------------------------------------------------------------------------------------------------------------------------------|
+| logical_plan      | Sort: home.time ASC NULLS LAST                                                                                                               |
+|                   |   TableScan: home projection=[co, hum, room, sensor, temp, time], full_filters=[home.time >= TimestampNanosecond(1688491380936276013, None)] |
+| physical_plan     | SortExec: expr=[time@5 ASC NULLS LAST]                                                                                                       |
+|                   |   EmptyExec: produce_one_row=false                                                                                                           |
+| Plan with Metrics | SortExec: expr=[time@5 ASC NULLS LAST], metrics=[output_rows=0, elapsed_compute=1ns, spill_count=0, spilled_bytes=0]                         |
+|                   |   EmptyExec: produce_one_row=false, metrics=[]
+```
+
+### Enable trace logging
+
+Use query tracing in {{% product-name %}} to get observability into a slow query.
+The tracing system follows the [OpenTelemetry traces](https://opentelemetry.io/docs/concepts/signals/traces/) model for providing observability into a request.
```

Comment:
Probably worth noting somewhere that adding trace headers to all queries, all the time, is not recommended.